source/base/subscriptor.cc

```diff
@@ -62,6 +62,22 @@ Subscriptor::Subscriptor (const Subscriptor &)
 {}
 
 
+
+#ifdef DEAL_II_WITH_CXX11
+Subscriptor::Subscriptor (Subscriptor &&subscriptor)
+  :
+  counter(0),
+  object_info (subscriptor.object_info)
+{
+  // Explicitly invoke the destructor of `Subscriptor` for the object
+  // to be moved from in order to guarantee that we're not moving an
+  // object that has subscriptions.
+  subscriptor.~Subscriptor();
```

Comment:
I don't think this is a good idea. The moved-from object must still continue to exist, and a destructor will in fact later be called for it. This way, it will be destroyed twice, which usually doesn't lead to anything good.

So I think you should copy the assertion instead.